lib/elixir/src/elixir_module.erl

```diff
@@ -85,7 +85,9 @@ compile(Line, Module, Block, Vars, E) ->
     (not elixir_config:get(bootstrap)) andalso
      'Elixir.Module':check_behaviours_and_impls(E, DataSet, DataBag, AllDefinitions, OverridablePairs),
 
-    CompileOpts = lists:flatten(bag_lookup_element(DataBag, {accumulate, compile}, 2)),
+    RawCompileOpts = bag_lookup_element(DataBag, {accumulate, compile}, 2),
+    DefinitionTuples = [Name || {Name, _, _, _} <- AllDefinitions],
```

Comment:
I would prefer to not do this here and instead use `lists:keyfind` when validating inline.

The rationale is that inline is used very rarely, so it doesn't make sense to always do this pass. Instead we do a lookup only when the inline attribute is used. keyfind is still in C, it should be fast enough.